File: Project_Sourcecode_go/state/CheckPin.go
package state

type CheckPin struct {
	State
}

func (s *CheckPin) IncorrectPin(max int) {
	if s.M.Attempts < max {
		s.M.Attempts++
		s.M.P.IncorrectPinMsg()
	} else {
		s.M.P.TooManyAttemptsMsg()
		s.M.ChangeState(1)
	}
}

func (s *CheckPin) CorrectPinAboveMin() {
	s.M.P.DisplayMenu()
	s.M.ChangeState(3)
}

func (s *CheckPin) CorrectPinBelowMin() {
	s.M.P.DisplayMenu()
	s.M.ChangeState(5)
}

func (s *CheckPin) Logout() {
	s.M.ChangeState(1)
}

func (s *CheckPin) GetName() string {
	return "CheckPin"
}

File: Project_Sourcecode_java/src/main/java/edu/iit/cs586/project/state/CheckPin.java
package edu.iit.cs586.project.state;

import edu.iit.cs586.project.MDA_EFSM;
import edu.iit.cs586.project.output.OP;

/**
 * @author CharmingLee <charminglee@jieztech.com>
 * @date 2022/5/1 22:27
 */
public class CheckPin extends State{
    public CheckPin(MDA_EFSM m) {
        super(m);
    }

    @Override
    public String getName() {
        return "CheckPin";
    }

    @Override
    public void logout() {
        this.m.changeState(1);
    }

    @Override
    public void incorrectPin(int max) {
        if (m.attempts < max) {
            m.attempts++;
            m.p.incorrectPinMsg();
        } else {
            m.p.incorrectPinMsg();
            m.p.tooManyAttemptsMsg();
            this.m.changeState(1);
        }
    }

    @Override
    public void correctPinAboveMin() {
        m.p.displayMenu();
        this.m.changeState(3);
    }

    @Override
    public void correctPinBelowMin() {
        m.p.displayMenu();
        this.m.changeState(5);
    }
}
